irr/include/irrString.h

```diff
@@ -176,10 +176,14 @@ class string
 		// no longer allowed!
 		_IRR_DEBUG_BREAK_IF((void *)c == (void *)c_str());
 
-		u32 len = calclen(c);
-		str.resize(len);
-		for (u32 l = 0; l < len; ++l)
-			str[l] = (T)c[l];
+        u32 len = calclen(c);
+        // We must check if the new length is different, otherwise we experience
+        //  a use-after-free. In such a case, we decide whether to resize based
+        //  on it's length.
```

Comment:
i think this doesn't explain it well
```suggestion
        // In case `c` is a pointer to our own buffer, we may not resize first
        // or it can become invalid.
```